### ISSUE TITLE
Midi Player bug w repeat endings and Dal Segno, DaCapo

### ISSUE THREAD

**Issue by BigHonkinDaddy** [powertab/powertabeditor#334]
i believe I've found a bug when using repeat endings and Da Capo or Dal Segno.  In the middle of my score, I have some measures that repeat (twice) but the last measure of the repeated section differs the second time around.  I used the repeat endings, and it worked great...the midi player correctly plays the measures twice, using the second ending the second time around.  great.  But then I have a Da Capo  (or even a Dal Segno) to return to the top (or to a Sign prior to the repeated measures).  The midi player correctly returns to the top (or to the sign) and continue playing...however, when it again gets to my repeated measures, it only plays those measures once and uses the second ending.  Its like the repeat counter for those measures doesn't get reset upon Da Capo or Dal Segno.

**Comment by psyomn** [comment]
Hey there, thank you for the report! 

Is there a sample file you can share that replicates this behavior? Doesn't need to be the actual song you're working on -- just something that behaves the same way. 

Thanks!

**Comment by BigHonkinDaddy** [comment]
yes... here is a sample, attached.  I couldn't figure out how to attach the .PT2 file, so just made a png out of it.  Hope that's okay.. if there's a way to attach the pt2 file, pls let me know.

By the way, I just realized that the repeat endings is beside the point.  if the Dal Segno brings the midi player back to before any repeated measures, those measures only play once.  if there are different endings to the repeat section (as in my score and the sample I provided), it will play just the last ending.

![PT Bug Dal Segno w Repeat endings](https://user-images.githubusercontent.com/77425146/131404678-17ce8e86-42da-4aa2-8a63-004e4efb32d3.png)

**Comment by cameronwhite** [comment]
Attaching my test file ([bug_334.zip](https://github.com/powertab/powertabeditor/files/7132891/bug_334.zip)). I think Github is a bit restrictive about file uploads, but zipping it up works
